docs/guides/Package-authoring-best-practices.md

```diff
@@ -0,0 +1,177 @@
+---
+title: Package authoring best practices
+description: A general guide of best practices for creating high quality NuGet packages.
+author: chgill-MSFT
+ms.author: chgill
+ms.date: 09/17/20
+ms.topic: conceptual
+---
+
+# Package authoring best practices
+
+This guidance is intended to give NuGet.org package authors a light-weight end to end reference for creating and publishing high quality packages. It will primarily focus on package-specific best practices such as metadata and packing. For more in depth suggestions for building high quality libraries, see the .NET [Open-source library guidance](https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/).
```

Comment:
should we rename the file, or move it to the server team sub-url, if it's nuget.org specific?  Should we split the guidelines into generic package authoring guidelines, and nuget.org specific ones?

Reply:
Would there be significant differences between guidelines for NuGet.org vs private packages?

Reply:
from a quick look:
* checking if a package id is used on nuget.org is unlikely to be relevant
* nuget.org's prefix reservation criteria is definitely not relevant
* authors matching the nuget.org account doesn't make sense
* license might not make sense/be relevant
* icon is still good advice, but probably much less likely and less important on a private feed than nuget.org. Particularly when an enterprise has a private feed per division/product/team.
* release notes may still be good advice, but it's generally really hard to do release notes, so in my opinion unlikely to actually get done.  In my previous team we pushed a new package after every commit, so the release notes would just be a duplicate of the git log. And often these packages weren't even for other teams to use, but a way to share code between different repos that make up a larger system (much like how nuget.org's source is split across multiple repos)

That's at least half the recommendations in this PR :)  I would hope that enterprise customers creating packages for internal feeds would think about each one and decide what's relevant to them. And for general best practices, maybe we should have pack warnings rather than docs, although that's very tricky because any new warning breaks people who treat warnings as errors, and they don't care that they asked for it, they still get angry about anything that breaks their build.

I'm not pushing for this, but it's a thought of mine. Several of the recommendations are nuget.org specific, and since nuget.org docs have their own subsection, does it below there?  If an enterprise customer is creating a package for proprietary, internal-only use, will they read the first few recommendations, see it's nuget.org specific, and stop reading the rest?   Or am I overthinking this and it's all fine as is?